### PR TITLE
Escape special characters in Jupyter circuit diagram

### LINKED ISSUE #3196: Glitched Circuit Diagram after flatten
**Description of the issue**
When a circuit has a parameterised gate as a product of symbols, flattening the circuit will produce a glitched circuit diagram where the parameterised expression is gone. This will further affect the circuit diagram if two qubit gates are also present in the circuit. 
**How to reproduce the issue**

```
import sympy as sp
from cirq import CNOT, rz, Circuit
cq = cirq.Circuit()
qubits = cirq.GridQubit.rect(2,1)
x = sp.symarray('x', 2)
cq.append([rz(x[0]*x[1])(qubits[-1]), CNOT(*tuple(qubits))])
new_cq, _= cirq.flatten(cq)
new_cq
```
This will product circuit diagram like this

This will product circuit diagram like this

(0, 0): ─────────────────────────@───
　　　　　　　　　　　│
(1, 0): ───Rz(pi*)───X───

**Cirq version**
0.8.2

### PR DESCRIPTION
**Pull request by abhik-99**
This PR fixes #3196 by using the ```html``` module for escaping characters when the circuit diagram is produced. The escaping allows easy and correct rendering of the diagram when it is used as a cell output of a Jupyter Notebook instead of using ```print(circuit)```.
Previsouly, the special characters like ```< >``` were not being rendered after the circuit was flattened as they were being interpreted as part of HTML tags. Using ```html.escape( )``` in ```circuit._repr_html(self)``` prevents that from happening.

### CODE REVIEW

**Comment by abhik-99** [comment]
Hey @cduck, please have a look at commit [c4e7292](https://github.com/quantumlib/Cirq/pull/3221/commits/c4e729263be1add71f66bb218282c4123a8e482f). Managed to pass the formatting test by [513cc4f](https://github.com/quantumlib/Cirq/pull/3221/commits/513cc4fce7d7deac6e9b64ced9de9173e3899d4d) but the test is written in the former.

File: cirq/circuits/circuit_test.py [hunk 1]
```diff
@@ -3817,3 +3818,49 @@ def test_zip():
...
+    circuit = cirq.Circuit([F2(a, b), F2(b, c), F2(a, c)])
+    original_diagram = """
+0: ─────< ' F ' > ────────────────< ' F ' > ───
+        │                         │
+1: ─────< ' F ' > ───< ' F ' > ───┼────────────
+                     │            │
+|c>: ────────────────< ' F ' > ───< ' F ' > ───
+""".strip()
+
+    escaped_diagram = ('<pre style="overflow: auto; '
```
Comment (balopat) [inline cirq/circuits/circuit_test.py:3851]: This is a little bit fragile to test against the full output string: for example, if we change the style tag, this test will have to change, which even though it is focusing on an unrelated logic. I would just simply assert that `_repr_html` contains `&lt; &#x27; F &#x27; &gt;`.

Comment (abhik-99) [reply cirq/circuits/circuit_test.py:3851]: Got it! Changing it.

Comment (abhik-99) [reply cirq/circuits/circuit_test.py]: Hey @balopat please have a look at commit [a45f2a0](https://github.com/quantumlib/Cirq/pull/3221/commits/a45f2a050e28c8ad1e9d7405e0820f9cd4c44c36).

File: cirq/circuits/circuit.py [hunk 2]
```diff
@@ -27,6 +27,8 @@
                     List, Optional, overload, Sequence, Set, Tuple, Type,
                     TYPE_CHECKING, TypeVar, Union)
 
+import html
+
```
Comment (cduck) [inline cirq/circuits/circuit.py:31]: Nit: No blank line needed here.

File: cirq/circuits/circuit_test.py [hunk 3]
```diff
@@ -3817,3 +3817,31 @@ def test_zip():
...
+        def __init__(self, n_qubits, label):
+            self.n_qubits = n_qubits
+            self.label = label
+
+        def num_qubits(self):
+            return self.n_qubits
+
+        def _circuit_diagram_info_(self, args):
+            return cirq.CircuitDiagramInfo(wire_symbols=[self.label] *
+                                           self.n_qubits)
```
Comment (cduck) [inline cirq/circuits/circuit_test.py:3835]: Keep this a simple as possible for the test.  You can remove the `__init__` method and have the other methods hard-code `2` and `"< ' F ' >"`.

File: cirq/circuits/circuit_test.py [hunk 4]
```diff
@@ -3817,3 +3817,31 @@ def test_zip():
...
+    a, b = cirq.LineQubit.range(2)
+    c = cirq.NamedQubit("|c>")
+
+    circuit = cirq.Circuit([F2(a, b), F2(b, c), F2(a, c)])
+
+    # Escaping Special Characters in Gate names.
+    assert '&lt; &#x27; F &#x27; &gt;' in circuit._repr_html_()
+
+    # Escaping Special Characters in Qubit names.
+    assert '|c&gt;' in circuit._repr_html_()
```
Comment (cduck) [inline cirq/circuits/circuit_test.py:3847]: The file is missing a trailing newline.  (Why is this not added by the auto formatter?)

Comment (abhik-99) [reply cirq/circuits/circuit_test.py]: Added one in.

File: cirq/circuits/circuit_test.py (hunk 3 above)
Comment (abhik-99) [reply cirq/circuits/circuit_test.py]: Yeah. @cduck please have a look at commit [bac6c2f](https://github.com/quantumlib/Cirq/pull/3221/commits/bac6c2f309a6d9cc9c9ba4ee64354ec6d635486b). Passed the format test in the last commit.

File: cirq/circuits/circuit.py (hunk 2 above)
Comment (abhik-99) [reply cirq/circuits/circuit.py]: Resolved in commit [54699e5](https://github.com/quantumlib/Cirq/pull/3221/commits/54699e5df1bc1c130eae57604ebfdc2dd5d9d07f)

File: cirq/circuits/circuit_test.py [hunk 5]
```diff
@@ -3817,3 +3817,29 @@ def test_zip():
...
         )
+
+
+def test_repr_html_escaping():
+
+    class TestGate(cirq.Gate):
+
+        def num_qubits(self):
+            self.n_qubits = 2
+            return self.n_qubits
```
Comment (balopat) [inline cirq/circuits/circuit_test.py:3828]: don't be afraid of hard-coding in tests. 

```
def num_qubits(self):          
            return 2
```

File: cirq/circuits/circuit_test.py [hunk 6]
```diff
@@ -3817,3 +3817,29 @@ def test_zip():
...
+
+        def _circuit_diagram_info_(self, args):
+            self.label = "< ' F ' >"
+            return cirq.CircuitDiagramInfo(wire_symbols=[self.label] *
+                                           self.n_qubits)
```
Comment (balopat) [inline cirq/circuits/circuit_test.py:3833]:
```
def _circuit_diagram_info_(self, args):
            return cirq.CircuitDiagramInfo(wire_symbols=["< ' F ' >", "< ' F ' >"])
```
same here

File: cirq/circuits/circuit_test.py (hunk 5 above)
Comment (abhik-99) [reply cirq/circuits/circuit_test.py]: Sure!

File: cirq/circuits/circuit_test.py (hunk 6 above)
Comment (abhik-99) [reply cirq/circuits/circuit_test.py]: Done.

**Pull request merged**